src/maybe.ts

```diff
@@ -34,6 +34,9 @@ export interface MaybeClasses<T> {
   /** Method variant for [`Maybe.mapOrElse`](../modules/_maybe_.html#maporelse) */
   mapOrElse<U>(this: Maybe<T>, orElseFn: (...args: any[]) => U, mapFn: (t: T) => U): U;
 
+  /** Method variant for [`Maybe.match`](../modules/_maybe_.html#match) */
+  match<T, U>(this: Maybe<T>, matchObj: { Just: (value: T) => U, Nothing: () => U }): U;
```

Comment:
- Can use the `Matcher` type from below.
- Also, let's match up the names and use `matcher` in all places